demo.scenario

```diff
@@ -0,0 +1,147 @@
+$ # Size of a Nix based docker image is somewhere between a Debian and Alpine
+$ # redis packaged in nixpkgs is not optimized to be minimal as is in Alpine
+$ # Let us fix this!
+$ bat redis-minimal.nix
+───────┬─────────────────────────────────────────────────────────────────────
+       │ File: redis-minimal.nix
+───────┼─────────────────────────────────────────────────────────────────────
+   1   │ { pkgs ? import <nixpkgs> {}
+   2   │ }:
+   3   │ pkgs.redis.overrideAttrs (old: {
+   4   │   # no need for systemd support in docker image
+   5   │   makeFlags = old.makeFlags ++ ["USE_SYSTEMD=no"];
+   6   │   # build static binary with musl
+   7   │   preBuild = ''
+   8   │     makeFlagsArray=(PREFIX="$out"
+   9   │                     CC="${pkgs.musl.dev}/bin/musl-gcc -static"
+  10   │                     CFLAGS="-I${pkgs.musl.dev}/include"
+  11   │                     LDFLAGS="-L${pkgs.musl.dev}/lib")
+  12   │   '';
+  13   │   # no need for other binaries then redis-server
+  14   │   postInstall = "rm -f $out/bin/redis-{benchmark,check-*,cli}";
+  15   │ })
+───────┴─────────────────────────────────────────────────────────────────────
+$ # In redis-minimal.nix we override the default redis build with 3 changes:
+$ # 1.) remove redis systemd support
+$ # 2.) build to static binary with musl
+$ # 3.) only redis-server binary is needed
+$ # Now all we need to is build docker image with our new minimal redis
+$ bat docker-redis-minimal.nix
+───────┬─────────────────────────────────────────────────────────────────────
+       │ File: redis-minimal.nix
+───────┼─────────────────────────────────────────────────────────────────────
+   1   │ { pkgs ? import <nixpkgs> {}
+   2   │ }:
+   3   │ let
+   4   │   redisMinimal = import ./redis-minimal.nix { inherit pkgs; };
+   5   │ in
+   6   │ pkgs.dockerTools.buildLayeredImage { # helper to build docker image
+   7   │   name = "nix-redis-minimal";        # give docker image a name
+   8   │   tag = "latest";                    # provide a tag
+   9   │   contents = [ redisMinimal ];       # use redisMinimal package
+  10   │ }
+───────┴─────────────────────────────────────────────────────────────────────
+$ # Lets build new docker image
+$ nix-build docker-redis-minimal.nix -o ./result 
+... SKIPPING OUTPUT ...
+/nix/store/83zcgs5xvzrgx09iv8s82wkabl8xkr03-docker-image-nix-redis-minimal.tar.gz
+$ docker load -i ./result
+Loaded image: nix-redis-minimal:latest
+$ docker images | grep redis
+debian-redis         latest    8366943c77e8    3 days ago      136MB
+alpine-redis         latest    aae644cd3417    3 days ago      6.99MB
+nix-redis            latest    30486183a209    50 years ago    45.4MB
+nix-redis-minimal    latest    a21238890680    50 years ago    2.02MB
+$ # Smaller then Alpine? Interesting!
+$ # Now we could claim Alpine is "bloated" :P, but we wont.
+$ # But we can not stop you from telling this to your friends :)
```

Comment:
Yeah. Let's leave this out :). We can leave the proof in the pudding.

Reply:
In case you leave this out, I would add least one sentence, so that the viewer has time to spot the smaller number in the output.

Reply:
I changed it a bit. Would this be better? /cc @grahamc @Mic92

Reply:
LGTM.